rust/template/d3log/src/dispatch.rs

```diff
@@ -0,0 +1,66 @@
+// dispatch is an external placeholder for a timely sink. It allows ports to be registered against
+// relations, and like forwarder, groups up sub-batches based on relation id and routes them
+// out the correct port. Used to hang management relation update ports off the broadcast tree
+
+use crate::{Batch, Error, Evaluator, Port, RecordBatch, Transport};
+
+use std::collections::HashMap;
+use std::sync::atomic::{AtomicUsize, Ordering};
+use std::sync::{Arc, Mutex};
+
+type DispatchMap = HashMap<String, Vec<(u64, Port)>>;
+
+#[derive(Clone)]
+pub struct Dispatch {
+    eval: Evaluator, // to be removed
+    count: Arc<AtomicUsize>,
+    handlers: Arc<Mutex<DispatchMap>>,
```

Comment:
Do we really need a `Mutex` as opposed to a `RwLock` here (and possibly in other places in the code)?